bin/bump_version.nim

```diff
@@ -0,0 +1,268 @@
+        while true:
+          stderr.write "increment [major], [m]inor, [p]atch (or [q]uit)? "
+          case stdin.readLine().toLowerAscii()
+          of "major":
+            inc result.major
+            result.minor = 0
+            result.patch = 0
+            break
+          of "m", "minor":
+            inc result.minor
+            result.patch = 0
+            break
+          of "p", "patch":
+            inc result.patch
+            break
+          of "q", "quit":
+            exit0("Quitting")
+          else:
+            stderr.writeLine "unrecognized choice. Try again."
+        let bumpedLine = line.replace(currentVersion, $result)
+        newContents.add bumpedLine
+      else:
+        newContents.add line
+      newContents.add "\n"
+
+    if result.major >= 0 and result.minor >= 0 and result.patch >= 0:
+      createBranchAndCheckout(result)
+      writeFile(nimblePath, newContents)
+    else:
+      error(&"could not detect version in nimble file: {nimblePath}")
+  else:
+    error(&"nimble file not found at path: {nimblePath}")
+
+proc bumpVersionFile(versionFile: string): Version =
+  ## Increments the version in the file at `versionFile`, and returns the new
+  ## version.
+  let pkgVersion = readFile(versionFile)
+  result = Version.init()
+  if pkgVersion.scanf("$i.$i.$i-$+.$i\n$.",
+                      result.major, result.minor, result.patch,
+                      result.pre, result.n):
+    inc result.n
+    createBranchAndCheckout(result)
+    writeFile(versionFile, $result & "\n")
+  else:
+    error(&"the version '{pkgVersion}' is not in the expected format.")
+
+proc bumpAndCommit: Version =
+  ## Creates a new branch that bumps the version in a `configlet.version` file
+  ## (if it exists), or the `configlet.nimble` file otherwise. Returns the new
+  ## version.
+  const
+    repoRootPath = currentSourcePath.parentDir().parentDir()
+    versionPath = repoRootPath / "configlet.version"
+
+  if fileExists(versionPath):
+    discard execAndCheck(GitLsFiles, ["--error-unmatch", versionPath],
+                         &"file is not tracked: {versionPath}")
+    result = bumpVersionFile(versionPath)
+    discard execAndCheck(GitAdd, [versionPath])
+  else:
+    let nimblePath = repoRootPath / "configlet.nimble"
+    result = bumpNimbleVersion(nimblePath)
+    discard execAndCheck(GitAdd, [nimblePath])
+
+  discard execAndCheck(GitCommit, ["-m", &"release: {result}"])
+  echo "Created release commit.\n"
+  echo execAndCheck(GitLog, ["-p", "-n1"])
+
+proc genPrBody: string =
+  ## Returns a suitable body for a release PR, listing every commit since the
+  ## previous release.
+  let previousTag = execAndCheck(GitDescribe, ["--abbrev=0"])
+  result = "Let's create a release for these commits:\n"
+  let commits = execAndCheck(GitLog, ["--oneline", &"{previousTag}..main"])
+  for line in commits.splitLines():
+    result.add "- "
+    # TODO: Avoid linking to many PRs due to contents of parentheses in
```

Comment:
```suggestion
    # TODO: Avoid linking too many PRs due to contents of parentheses in
```

Reply:
Both are gramatically correct, but the original "to" was redundant - it could've been "avoid linking many PRs".

I've reworded and explained a bit better with https://github.com/exercism/configlet/pull/529/commits/fdac79490844e34b0fa2d9d452d97b92d325e7f3.